balancer-js/src/modules/pools/index.spec.ts

```diff
@@ -10,6 +10,33 @@ import {
 import { Pools } from './';
 import { BalancerSDK } from '../sdk.module';
 import { expect } from 'chai';
+import nock from 'nock';
```

Comment:
fetch vs axios - we have fetch now, so let's just use it.
for:
* this PR is adding a lot of new things, so we can mark it as a breaking release (needing node v18 update)
* SDK is not yet that popular and adding legacy now will make it longer to transition out.
* slimmer dependency tree
against:
* existing node client will need to update node